app/src/main/kotlin/com/intive/picover/parties/view/PartiesScreen.kt

```diff
@@ -56,8 +56,8 @@ fun PartiesScreen(
 		MVIState.Type.LOADING -> PicoverLoader(modifier = Modifier.fillMaxSize())
 		MVIState.Type.LOADED -> LoadedContent(
 			parties = state.parties,
-			onPartyClick = viewModel::onPartyClick,
-			onFabClick = viewModel::onAddPartyClick,
+			onPartyClick = { viewModel.emitEffect(NavigateToPartyDetails(it)) },
```

Comment:
I'm thinking about it and in that case the VM is totally redundant part of the navigation, we are calling `emitEffect` to just react to that effect and navigate. We could directly call `onPartClick =  { navController.navigate("partyDetails/${effect.partyId}") }`, because VM is not taking any part in the side-effect handling. Maybe it's only like that in the simplest cases like this, I'm just loudly thinking. And of course that is not related to your changes because it was even more complicated in the previous approach.

Reply:
Good point @BorysChajdas. I think that probably the best idea will be to use emitEffect here when there will be some sort of a tested navigator in VM. At this moment for this particular situation i have the same feelings.

On the other hand, i am not sure how to do it properly and clean, so the solution with emitting effects and navigation in screen is also ok for me.

Reply:
I agree in this case the vm is redundant, but if we go with your idea then the ui will keep part of the logic (what should happen when a user clicks on something).

Reply:
I would suggest to create some sort of "processor" to handle navigation events. Note that current handling of navigation from side effects is done in @Composable scope. Now let's think about testing, you have to engage compose framework to execute such test, but `navController::navigate` call is not related to, so there is a high coupling between these to - it should be removed.